services/brig/src/Brig/API/Public.hs

```diff
@@ -1417,15 +1466,22 @@ checkHandlesH (_ ::: _ ::: req) = do
   free <- lift $ API.checkHandles handles (fromRange num)
   return $ json (free :: [Handle])
 
-getHandleInfoUnqualifiedH :: UserId -> Handle -> Handler Public.UserHandleInfo
+getHandleInfoUnqualifiedH :: UserId -> Handle -> Handler (Union '[WithStatus 200 Public.UserHandleInfo, HandleNotFound])
 getHandleInfoUnqualifiedH self handle = do
   domain <- viewFederationDomain
-  Public.UserHandleInfo . Public.profileQualifiedId <$> getUserByHandleH self domain handle
-
-getUserByHandleH :: UserId -> Domain -> Handle -> Handler Public.UserProfile
-getUserByHandleH self domain handle =
-  ifNothing (notFound "handle not found")
-    =<< getHandleInfo self (Qualified handle domain)
+  profile <- getUserByHandleH self domain handle
+  case profile of
+    Z (I (WithStatus userProfile)) ->
+      Servant.respond . WithStatus @200 . Public.UserHandleInfo . Public.profileQualifiedId $ userProfile
+    S (Z (I x)) -> Servant.respond x
+    S (S x) -> case x of
```

Comment:
This `S (X (I x))` is very confusing syntax. I'd be in favour of either removing this complexity to leak into these method bodies here. Having slightly improved swagger is not worth the tradeoff to have to write this kind of non-intuitive code in my opinion. Also, it's unclear why the profile is converted to a UserHandleInfo here. This is confusing and needs at least some comments. Or instead why not call another Data.lookupfunction that was there before?

Reply:
Oh yeah, I wanted to ask @arianvp and @fisx if they know of a better way to do this.

Reply:
Removed it, let's talk about what it would take for UVerb to be ready.